wlauto/workloads/reader/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,238 @@
+/*    Copyright 2014-2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arm.wlauto.uiauto.reader;
+
+import android.os.Bundle;
+
+// Import the uiautomator libraries
+import com.android.uiautomator.core.UiObject;
+import com.android.uiautomator.core.UiObjectNotFoundException;
+import com.android.uiautomator.core.UiSelector;
+
+import com.arm.wlauto.uiauto.UxPerfUiAutomation;
+
+import static com.arm.wlauto.uiauto.BaseUiAutomation.FindByCriteria.BY_ID;
+import static com.arm.wlauto.uiauto.BaseUiAutomation.FindByCriteria.BY_TEXT;
+import static com.arm.wlauto.uiauto.BaseUiAutomation.FindByCriteria.BY_DESC;
+
+import java.util.concurrent.TimeUnit;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class UiAutomation extends UxPerfUiAutomation {
+
+    protected Bundle parameters;
+    protected String packageName;
+    protected String packageID;
+
+    private long networkTimeout =  TimeUnit.SECONDS.toMillis(20);
+    private long searchTimeout =  TimeUnit.SECONDS.toMillis(20);
+
+    public void runUiAutomation() throws Exception {
+        parameters = getParams();
+        packageName = parameters.getString("package");
+        packageID = packageName + ":id/";
+
+        String filename = parameters.getString("filename").replace("0space0", " ");
+        String[] searchStrings =
+            parameters.getString("search_string_list").replace("0space0", " ").split("0newline0");
+
+        setScreenOrientation(ScreenOrientation.NATURAL);
+
+        dismissWelcomeView();
+        openFile(filename);
+        gesturesTest();
+        // Wait one second then tap the centre to bring up the menu gui
+        sleep(1);
```

Comment:
The comment should explain why the wait is necessary. Also, is 1 second going to be enough for slower devices?